testsuite/standalone/src/test/java/org/wildfly/extras/creaper/core/online/SslOnlineManagementClientTest.java

```diff
@@ -0,0 +1,157 @@
+import org.wildfly.extras.creaper.core.ManagementClient;
+import org.wildfly.extras.creaper.core.offline.OfflineManagementClient;
+import org.wildfly.extras.creaper.core.offline.OfflineOptions;
+import org.wildfly.extras.creaper.security.KeyPairAndCertificate;
+import org.wildfly.extras.creaper.test.ManualTests;
+import org.wildfly.extras.creaper.test.WildFlyTests;
+
+/**
+ * This test <b>needs</b> a manually-controlled Arquillian container.
+ *
+ * <p>WildFly test. For AS7, there is no way to have SSL only secured management interface and to manage the server
+ * with Arquillian.</p>
+ *
+ * @see org.wildfly.extras.creaper.core.online.operations.admin.ReloadWhenRestartRequiredTest
+ */
+@RunAsClient
+@Category({ManualTests.class, WildFlyTests.class})
+public class SslOnlineManagementClientTest extends OnlineManagementClientTest {
+
+    private static final String SECURITY_REALM = "SslSecuredTestRealm";
+
+    private static final String SERVER_KEY_ALIAS = "serverKeyAlias";
+    private static final String SERVER_KEY_PASSWORD = "serverKeyPassword";
+    private static final String SERVER_CERTIFICATE_ALIAS = "serverCertificateAlias";
+    private static final String SERVER_CERTIFICATE_PASSWORD = "serverCertificatePassword";
+    private static final String CLIENT_KEY_ALIAS = "clientKeyAlias";
+    private static final String CLIENT_KEY_PASSWORD = "clientKeyPassword";
+    private static final String CLIENT_CERTIFICATE_ALIAS = "clientCertificateAlias";
+    private static final String CLIENT_CERTIFICATE_PASSWORD = "clientCertificatePassword";
+
+    private static final ConfigurationFileBackup CONFIGURATION_BACKUP = new ConfigurationFileBackup();
+    private static OfflineManagementClient offlineClient;
+
+    private static File serverKeyStoreFile;
+    private static File serverTrustStoreFile;
+    private static File clientKeyStoreFile;
+    private static File clientTrustStoreFile;
+    private static SslOptions sslOptions;
+
+
+    @ArquillianResource
+    private ContainerController controller;
+
+
+    @BeforeClass
+    public static void prepareAndConfigure() throws IOException, CommandFailedException, GeneralSecurityException {
+        offlineClient = ManagementClient.offline(OfflineOptions.standalone().rootDirectory(new File("target/jboss-as"))
+                        .configurationFile("standalone.xml").build());
+
+
+        offlineClient.apply(CONFIGURATION_BACKUP.backup());
+
+        prepareKeyAndTrustStoreFiles();
+        prepareSslOptions();
+        configureServerManagementInterfaces();
+    }
+
+    @AfterClass
+    public static void restoreConfiguration() throws CommandFailedException {
+        offlineClient.apply(CONFIGURATION_BACKUP.restore());
+    }
+
+    @AfterClass
+    public static void deleteKeyAndTrustStoreFiles() {
+        serverKeyStoreFile.delete();
+        serverTrustStoreFile.delete();
+        clientKeyStoreFile.delete();
+        clientTrustStoreFile.delete();
```

Comment:
Funny, I remember calls like this to not pass, because there's an unused return value. If it passes, the better :-)